pennylane_orquestra/orquestra_device.py

```diff
@@ -382,3 +387,254 @@ def qubit_operator_string(self, observable):
         # Use consecutive integers as default wire_map
         wire_map = {v: idx for idx, v in enumerate(self.wires)}
         return _terms_to_qubit_operator_string(coeffs, obs_list, wires=wire_map)
+
+    def execute(self, circuit, **kwargs):
+        # Input checks
+        not_all_expval = any(obs.return_type is not Expectation for obs in circuit.observables)
+        if not_all_expval:
+            raise NotImplementedError(
+                f"The {self.short_name} device only supports returning expectation values."
+            )
+
+        self.check_validity(circuit.operations, circuit.observables)
+
+        # 1. Create qasm strings from the circuits
+        try:
+            qasm_circuit = self.serialize_circuit(circuit)
+        except AttributeError:
+            # QuantumTape case: need to extract the CircuitGraph
+            qasm_circuit = self.serialize_circuit(circuit.graph)
+
+        # 2. Create the qubit operators
+        ops, identity_indices = self.process_observables(circuit.observables)
+
+        if not ops:
+            # All the observables were identity, no workflow submission needed
+            return self._asarray([1] * len(identity_indices))
+
+        ops_json = json.dumps(ops)
+
+        # Single step: need to nest the operators into a list
+        ops = [ops_json]
+        qasm_circuit = [qasm_circuit]
+
+        # 4-5. Create the backend specs & workflow file
+        workflow = gen_expval_workflow(
+            self.qe_component,
+            self.backend_specs,
+            qasm_circuit,
+            ops,
+            resources=self._resources,
+            **kwargs,
+        )
+        file_id = str(uuid.uuid4())
+        filename = f"expval-{file_id}.yaml"
+        filepath = write_workflow_file(filename, workflow)
+
+        # 6. Submit the workflow
+        workflow_id = qe_submit(filepath, keep_file=self._keep_files)
+
+        if self._keep_files:
+            self._filenames.append(filename)
+
+        self._latest_id = workflow_id
+
+        # 7. Loop until finished
+        results = self.single_step_results(workflow_id)
+
+        # Insert the theoretical value for the expectation value of the
+        # identity operator
+        for idx in identity_indices:
+            results.insert(idx, 1)
+
+        res = self._asarray(results)
+        return res
+
+    def single_step_results(self, workflow_id):
+        """Extracts the results of a single step obtained for a workflow.
+
+        This method assumes that the workflow had a single step and that the
+        structure of the result corresponds to results sent by Orquestra API
+        v1.0.0.
```

Comment:
Is there a way to verify this? Perhaps the API version can be checked, and a warning emitted if it is higher than 1.0.0?

Reply:
Hmm, good question! `qe --version` seems to output only the `Client` and `Server` versions :thinking: